pkg/proxy/winkernel/proxier.go

```diff
@@ -128,12 +128,21 @@ type endpointsInfo struct {
 	refCount   uint16
 }
 
+//Uses mac prefix and IP address to return a mac address
+//This ensures mac addresses are unique for proper load balancing
+func conjureMac(macPrefix string, ip net.IP) string { 
+	p := ip
+	p4 := p.To4();
+	a, b, c, d := p4[0], p4[1], p4[2], p4[3]
+	return fmt.Sprintf("%v-%02x-%02x-%02x-%02x", macPrefix, a, b, c, d) 
+}
+
 func newEndpointInfo(ip string, port uint16, isLocal bool) *endpointsInfo {
 	info := &endpointsInfo{
 		ip:         ip,
 		port:       port,
 		isLocal:    isLocal,
-		macAddress: "00:11:22:33:44:55", // Hardcoding to some Random Mac
+		macAddress: conjureMac("00-11", net.ParseIP(ip)), 
```

Comment:
The low 2 bits of the first octet of a MAC address have reserved meanings; in theory, you should have the 0x02 bit set, indicating that it's "locally administered" rather than "globally unique". (And you should not have the 0x01 bit set since that would indicate that it was a multicast address rather than unicast.)

Reply:
Updated in latest commit